packages/widget/src/hooks/useCreateCosmosWallets.tsx

```diff
@@ -159,37 +56,83 @@ export const useCreateCosmosWallets = () => {
-        const initialChainIds = (() => {
-          if (isWC) return walletConnectMainnetChainIdsInitialConnect;
-          if (wallet === WalletType.OKX) return okxWalletChainIdsInitialConnect;
-          if (wallet === WalletType.KEPLR) {
-            if (mobile) {
-              return keplrMainnetChainIdsInitialConnect.filter(
-                (chainId) => chainId !== "wormchain",
-              );
-            }
-            return keplrMainnetChainIdsInitialConnect;
-          }
-          return walletMainnetChainIdsInitialConnect;
-        })().filter(
-          (x) =>
-            chains
-              ?.filter((z) => z.chainType === ChainType.Cosmos)
-              .map((y) => y.chainID)
-              .includes(x) && mainnetChains.map((c) => c.chainId).includes(x),
-        );
-        const connectEco = async () => {
+        const initialChainIds = filterValidChainIds(getInitialChainIds(wallet), chains);
+
+        const updateSourceWallet = () => {
+          setCosmosWallet({
+            walletName: wallet,
+            chainType: ChainType.Cosmos,
+          });
+        };
+
+        const connectWallet = async ({
+          chainIdToConnect,
+          shouldUpdateSourceWallet = true,
+        }: {
+          chainIdToConnect?: string;
+          shouldUpdateSourceWallet?: boolean;
+        }) => {
+          const connectToInitialChainId =
+            !chainIdToConnect || (chainIdToConnect && initialChainIds.includes(chainIdToConnect));
           try {
-            await connect({
-              chainId: initialChainIds,
+            if (chainIdToConnect) {
+              const chainInfo = getChainInfo(chainIdToConnect);
+              if (!chainInfo)
+                throw new Error(`connect: Chain info not found for chainID: ${chainId}`);
+              if (!mobile && !isWC) {
+                await getWallet(wallet).experimentalSuggestChain(chainInfo);
+              }
+            }
+
+            const response = await connect({
+              chainId: connectToInitialChainId ? initialChainIds : chainIdToConnect,
               walletType: wallet,
               autoReconnect: false,
             });
+
+            if (!response?.accounts) {
+              throw new Error("failed to get accounts from wallet");
+            }
+
+            if (sourceAsset === undefined) {
+              const chain = chains?.find((x) => x.chainID === "cosmoshub-4");
+              const asset = assets?.find((x) => x.denom === "uatom");
+              setSourceAsset({
+                chainID: chain?.chainID,
+                chainName: chain?.chainName,
+                ...asset,
+              });
+            }
+
+            const chainIdToAddressMap = Object.fromEntries(
+              await Promise.all(
+                initialChainIds.map(async (chainId) => [
+                  chainId,
+                  (await getWallet(wallet).getKey(chainId)).bech32Address,
+                ]),
+              ),
+            );
```

Comment:
graz's `connect` already have the address we need, don't need to fetch `getKey` anymore https://github.com/graz-sh/graz/blob/dev/packages/graz/src/actions/account.ts#L103